R/between.R

```diff
@@ -4,22 +4,45 @@ between <- function(x,lower,upper,incbounds=TRUE) {
   if (is.logical(lower)) lower = as.integer(lower)   # typically NA (which is logical type)
   if (is.logical(upper)) upper = as.integer(upper)   # typically NA (which is logical type)
   is_strictly_numeric <- function(x) is.numeric(x) && !inherits(x, "integer64")
+  try_posix_cast <- function(x, tz) {
+    x_time = as.POSIXct(x, tz = tz)
+    if (is.na(x_time)) return(x)
+    return(x_time)
+  }
+  if (inherits(x, "POSIXct")) {
+    # allow fast between on POSIX vs non-explicit-POSIX in some circumstances
+    #   (biggest worry is hard-to-predict timezone mismatch)
+    if (!is.null(tz <- attr(x, 'tzone'))) {
+      lower = try_posix_cast(lower, tz)
+      upper = try_posix_cast(upper, tz)
```

Comment:
Rather than casting at R level, couldn't between.c handle both integer64 and POSIXct at C level.  My first thought is that if x, upper and lower aren't all POSIXct and all in the same timezone, there should at least be a warning.

Reply:
Could be, but handling character on input of `Cbetween` will significantly increase its code complexity. I think it make sense to leave coercion (at least from character type) in R. We need to wrap it with `tryCatch`, as you noticed in another comment, there are no tests for failed coercion of character to POSIXct.

Reply:
I added error, rather than warning. If we want to have warning instead I would prefer to drop timezone from all arguments, warn, and proceed. But error should be OK, tz mismatch should be aligned on user side.

Reply:
int64 support is still TODO